packages/toolpad-core/src/Crud/List.tsx

```diff
@@ -0,0 +1,450 @@
+        url.searchParams.set('sort', JSON.stringify(sortModel));
+      } else {
+        url.searchParams.delete('sort');
+      }
+
+      if (!appWindow.frameElement) {
+        appWindow.history.pushState({}, '', url);
+      }
+    }
+  }, [appWindow, sortModel]);
+
+  const loadData = React.useCallback(async () => {
+    setError(null);
+    setIsLoading(true);
+    try {
+      const listData = await getMany({
+        paginationModel,
+        sortModel,
+        filterModel,
+      });
+      setRowsState({
+        rows: listData.items,
+        rowCount: listData.itemCount,
+      });
+    } catch (listDataError) {
+      setError(listDataError as Error);
+    }
+    setIsLoading(false);
+  }, [filterModel, getMany, paginationModel, sortModel]);
+
+  React.useEffect(() => {
+    loadData();
+  }, [filterModel, getMany, loadData, paginationModel, sortModel]);
+
+  const handleRefresh = React.useCallback(() => {
+    if (!isLoading) {
+      loadData();
+    }
+  }, [isLoading, loadData]);
+
+  const handleRowClick = React.useCallback<GridEventListener<'rowClick'>>(
+    ({ row }) => {
+      if (onRowClick) {
+        onRowClick(row.id);
+      }
+    },
+    [onRowClick],
+  );
+
+  const handleItemEdit = React.useCallback(
+    (itemId: DataModelId) => () => {
+      if (onEditClick) {
+        onEditClick(itemId);
+      }
+    },
+    [onEditClick],
+  );
+
+  const handleItemDelete = React.useCallback(
+    (itemId: DataModelId) => async () => {
+      const confirmed = await dialogs.confirm(`Do you wish to delete item "${itemId}"?`, {
+        title: 'Delete item?',
+        severity: 'error',
+        okText: 'Delete',
+        cancelText: 'Cancel',
+      });
+
+      if (confirmed) {
+        setIsLoading(true);
+        try {
+          await deleteOne?.(itemId);
+
+          if (onDelete) {
+            onDelete(itemId);
+          }
+
+          notifications.show('Item deleted successfully.', {
+            severity: 'success',
+            autoHideDuration: 3000,
+          });
+          loadData();
+        } catch (deleteError) {
+          notifications.show(`Failed to delete item. Reason: ${(deleteError as Error).message}`, {
+            severity: 'error',
+            autoHideDuration: 3000,
+          });
+        }
+        setIsLoading(false);
+      }
+    },
+    [deleteOne, dialogs, loadData, notifications, onDelete],
+  );
+
+  const DataGridSlot = slots?.dataGrid ?? DataGrid;
+
+  const initialState = React.useMemo(
+    () => ({
+      pagination: { paginationModel: { pageSize: initialPageSize } },
+    }),
+    [initialPageSize],
+  );
+
+  const columns = React.useMemo<GridColDef[]>(() => {
+    return [
+      ...(fields.map((field) =>
+        field.type === 'longString' ? { ...field, type: 'string' } : field,
+      ) as GridColDef[]),
+      {
+        field: 'actions',
+        type: 'actions',
+        flex: 1,
+        align: 'right',
+        getActions: ({ id }) => [
```

Comment:
A pinned column would be very useful for these actions but is not a feature of the basic data grid.
Open for discussion, the current behavior might not be ideal.